src/realm_dart_sync.cpp

```diff
@@ -26,24 +27,25 @@ RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, realm_h
     struct request_copy_buf {
         std::string url;
         std::string body;
-        std::map<std::string, std::string> headers;
-        std::vector<realm_http_header_t> headers_vector;
+        std::vector<std::pair<std::string, std::string>> headers_values;
+        std::vector<realm_http_header_t> headers;
     } buf;
 
     buf.url = request.url;
     buf.body = std::string(request.body, request.body_size);
-    buf.headers_vector.reserve(request.num_headers);
+    buf.headers_values.reserve(request.num_headers);
+    buf.headers.reserve(request.num_headers);
     for (size_t i = 0; i < request.num_headers; i++) {
-        auto [it, _] = buf.headers.emplace(request.headers[i].name, request.headers[i].value);
-        buf.headers_vector.push_back({ it->first.c_str(), it->second.c_str() });
+        auto& ref = buf.headers_values.emplace_back(request.headers[i].name, request.headers[i].value);
+        buf.headers.push_back({ ref.first.c_str(), ref.second.c_str() });
```

Comment:
```suggestion
        auto& [name, value] = buf.headers_values.emplace_back(request.headers[i].name, request.headers[i].value);
        buf.headers.push_back({ name.c_str(), value.c_str() });
```

I find this clearer. Your choice if you want to make this change. This was mostly an FYI in case you didn't know it was possible.

Reply:
nice! I will use it